workflows/postprocessing.nf

```diff
@@ -81,7 +85,7 @@ workflow POSTPROCESSING {
     .collectFile(storeDir: "${params.outdir}/pipeline_info/configs",cache: false)
 
     writemeta()
-    filtered = EXCLUDE_MNPS(ch_samplesheet).ch_output_excludemnps
+    def ch_output_from_excludemnps = EXCLUDE_MNPS(ch_samplesheet).ch_output_excludemnps
```

Comment:
ça marche, ça? De mon côté, j'avais souvent des problèmes en utilisant def sur des appels de fonctions

Reply:
Oui ça fonctionne. J'ai eu des problèmes de compilations au début quand j'ai commencé les changements. Je voyais des erreurs comme ça:
`Variable `vcfWithTags` already defined in the process scope @ line 118, column 32.`

Ça semble être un problème connu qui se produit dans un workflow. J'ai trouvé des messages de gens qui ont eu ce problème. Entre autres: https://midnighter.github.io/nextflow-gotchas/gotchas/variable-scope/#mixing-variables (voir mixing-variables).

Reply:
J'ai réglé en enlevant toutes les variables globales pas nécéssaires de mon côté. 

Je crois que souvent les gens font l'inverse. Je crois que c'est parce qu'ils se sensibilisent à la question sur le tard et que, justement à cause de ce problème, ils sont forcés de tout changer d'un coup pour que le code compile et ils abandonnent.

En plus, la documentation de nextflow n'est pas claire sur les bonnes pratiques à suivre de ce côté.  Dans les exemples qu'ils donnent, ils ne prennent souvent pas la peine de mettre les `def`.  J'ai trouvé cette note cependant:
`The use of global variables and params in named workflows is discouraged because it breaks the modularity of the workflow. As a best practice, every workflow input should be explicitly defined as such in the take: block, and params should only be used in the implicit workflow.`

Reply:
Ça se peut que tu as eu un autre problème de ton côté. Il serait peut-être aussi reporté dans ce document aussi?
https://midnighter.github.io/nextflow-gotchas/gotchas/variable-scope/

Reply:
Je vois rien qui ressemble à ce que tu dis. Il y a 3 gotchas. Le premier n'en est pas vraiment un selon moi ... Le deuxième est je crois réglé parce que nous on n'a pas ce problème. Le troisième est le bug que j'ai décris.